sdk/ml/azure-ai-ml/azure/ai/ml/entities/_indexes/utils/_open_ai_utils.py

```diff
@@ -0,0 +1,32 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+from typing import Optional
+import re
+
+from  azure.ai.ml._utils._arm_id_utils import is_ARM_id_for_resource
+from azure.ai.ml._scope_dependent_operations import OperationScope
+
+OPEN_AI_PROTOCOL_TEMPLATE = "azure_open_ai://deployment/{}/model/{}"
+OPEN_AI_PROTOCOL_REGEX_PATTERN = OPEN_AI_PROTOCOL_TEMPLATE.format(".*", ".*")
+
+def build_open_ai_protocol(s: Optional[str] = None):
+    if not s or re.match(OPEN_AI_PROTOCOL_REGEX_PATTERN, s, re.IGNORECASE):
+        return s
+    else:
+        return OPEN_AI_PROTOCOL_TEMPLATE.format(s, s)
```

Comment:
Why do we assume/force the deployment the same as the model name?

Reply:
good catch, this was from the old sdk where we want to save some typing for customers if their model name and deployment name are the same. I will change this.